input/script_utils_test.go

```diff
@@ -2034,3 +2035,212 @@ func TestSpecificationKeyDerivation(t *testing.T) {
 			actualRevocationPrivKeyHex)
 	}
 }
+
+// BenchmarkScriptBuilderAlloc benchmarks the script builder's default
+// allocation.
+func BenchmarkScriptBuilderAlloc(t *testing.B) {
+	dummyData := []byte("dummy data")
+	randomPriv, err := btcec.NewPrivateKey()
+	require.NoError(t, err)
+	randomPub := randomPriv.PubKey()
+	randomPubBytes := randomPub.SerializeCompressed()
+
+	t.ReportAllocs()
+	t.ResetTimer()
+
+	for i := 0; i < t.N; i++ {
+		for j := 0; j < 1000; j++ {
```

Comment:
I don't see the need to the inner (`j`) loop here. Couldn't we just use:
`for i := 0; i < t.N * 1000; i++ {` ?

Reply:
Yeah, I guess that makes sense. It's just that `for i := 0; i < t.N; i++ {` is a quite common pattern in Golang benchmark tests that I wanted the fact that we do 1k iterations per benchmark iteration stand out a bit more. But a times thousand also makes sense, updated.